### DISCUSSION TITLE
Issues implementing the plugin in flutter

### DISCUSSION THREAD

**Discussion by Aagam9833** [woodemi/quick_usb#47] [Q&A]
I am trying to use the plugin in my flutter application. So to test it out I used the example code. The plugin has been able to recognize device name, VID, PID and other description. However when I tap on BulkTransferIn/Out it throws the following exception

![quickusb](https://user-images.githubusercontent.com/64311579/152162374-03dba7f4-440e-42ea-8388-90ee8a511be4.png)

What is the issue I don't understand. I would really appreciate if anyone can help me.

P.s : I am working on my company's custom device, and they have native java code with USBManager class wherein  they can connect a custom device in charging port and read data from it. But now they wanna switch to flutter and I can't find any plugin which can help me with this issue. Any help in this regards is also highly appreciated

**Comment by Sunbreak** [comment]
https://github.com/woodemi/quick_usb/blob/57fdf0f596ef330de4337837168d7cc20dec6426/android/src/main/kotlin/com/example/quick_usb/QuickUsbPlugin.kt#L141

It seems to be `buffer.count()` throwing the exception. You'd better debug with internal Android SDK/logic code
